tests/integration/kafka/test_producer.py

```diff
@@ -71,3 +74,58 @@ def test_partitions_for(self, producer: KafkaProducer, new_topic: NewTopic) -> N
         assert partitions[0].id == 0
         assert partitions[0].replicas == [1]
         assert partitions[0].isrs == [1]
+
+
+@pytest.fixture(scope="function", name="aioproducer")
+async def fixture_aioproducer(
+    kafka_servers: KafkaServers,
+    loop: asyncio.AbstractEventLoop,
+) -> Iterator[AIOKafkaProducer]:
+    try:
+        aioproducer = AIOKafkaProducer(bootstrap_servers=kafka_servers.bootstrap_servers, loop=loop)
+        yield aioproducer
+    finally:
+        await aioproducer.stop()
+
+
+class TestAsyncSend:
+    async def test_async_send(self, aioproducer: AIOKafkaProducer, new_topic: NewTopic) -> None:
+        key = b"key"
+        value = b"value"
+        partition = 0
+        timestamp = int(time.time() * 1000)
+        headers = [("something", b"123"), (None, "foobar")]
+
+        aiofut = await aioproducer.send(
+            new_topic.topic,
+            key=key,
+            value=value,
+            partition=partition,
+            timestamp=timestamp,
+            headers=headers,
+        )
+        message = await aiofut
+
+        assert message.offset() == 0
+        assert message.partition() == partition
+        assert message.topic() == new_topic.topic
+        assert message.key() == key
+        assert message.value() == value
+        assert message.timestamp()[0] == Timestamp.CREATE_TIME
+        assert message.timestamp()[1] == timestamp
```

Comment:
suggestion (non-blocking): Should we check the headers?

Reply:
`send` uses the `confluent_kafka.Producer.produce` method, which has a bit of a caveat [returning headers on a Message](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Producer.produce):

> Currently message headers are not supported on the message returned to the callback. The msg.headers() will return None even if the original message had headers set.

Reply:
Gotcha ✅